### PR TITLE
Shuffleboard's getEntry() now returns a GenericEntry

### PR DESCRIPTION
**Pull request by calcmogul**

### CODE REVIEW

**Comment by SamCarlberg** [comment]
A few other examples are showing the old API, too:

https://github.com/wpilibsuite/frc-docs/blob/c1a5a45e8a1de0fecbbbe51ea2d5984af346a5a1/source/docs/software/dashboards/shuffleboard/layouts-with-code/sending-data.rst?plain=1#L35-L37

https://github.com/wpilibsuite/frc-docs/blob/c1a5a45e8a1de0fecbbbe51ea2d5984af346a5a1/source/docs/software/dashboards/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst?plain=1#L46

https://github.com/wpilibsuite/frc-docs/blob/c1a5a45e8a1de0fecbbbe51ea2d5984af346a5a1/source/docs/software/dashboards/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst?plain=1#L61

**Comment by calcmogul** [comment]
I have no idea if the C++ version is correct. The API returns a pointer, and NetworkButton takes a value.

**Review by SamCarlberg — APPROVED** [review]

File: source/docs/software/dashboards/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst [hunk 1]
```diff
@@ -58,7 +58,7 @@ The following example demonstrates how to create a button on your dashboard that
   .. code-block:: c++
 
     frc::ShuffleboardTab& tab = frc::Shuffleboard::GetTab("Shooter");
-    nt::NetworkTableEntry shooterEnable = tab.Add("Shooter Enable", false).GetEntry();
+    nt::GenericEntry& shooterEnable = *tab.Add("Shooter Enable", false).GetEntry();
```
Comment (Starlight220) [inline source/docs/software/dashboards/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst:61]: Doesn't this return a reference rather than a pointer?

Comment (calcmogul) [reply source/docs/software/dashboards/shuffleboard/advanced-usage/shuffleboard-tuning-pid.rst:61]: It doesn't.
```
shuffleboard/SimpleWidget.h
33:  nt::GenericEntry* GetEntry();
42:  nt::GenericEntry* GetEntry(std::string_view typeString);
```

**Review by sciencewhiz — APPROVED** [review]

**Pull request merged**